src/include/migraphx/op/scatter_op.hpp

```diff
@@ -67,38 +72,60 @@ struct scatter_op : op_name<Derived>
         // If non-packed, this converts to a packed output while preserving permutation of tensor
         return inputs.front().with_lens(inputs.front().lens());
     }
+   
+    // iterate through items in shape
+    template <class TensorView0, class TensorView1>
+    void scatter_reduce_iterate(TensorView0 indices, TensorView1 output, TensorView1 update) const
+    {
+        auto ind_s = indices.get_shape();
+        auto output_shape = output.get_shape();
+        auto axis_dim_size = output_shape.lens()[axis];
+        shape_for_each(ind_s, [&](const auto& idx) {
+                auto out_idx = idx;
+
+                // tensor_view::() invokes indexing logic of
+                // std::size_t shape::index(std::size_t i) const
+                auto index = indices(idx.begin(), idx.end());
+
+                // this addition doesn't necessarily make index positive if index was out of bounds
+                index         = (index < 0) ? index + axis_dim_size : index;
+                if(skip_out_of_bounds and index < 0)
+                {
+                    return;
+                }
+                else
+                {
+                    assert(index >= 0);
+                }
+                out_idx[axis] = index;
+                // skip index out of bounds if attribute on, else assert
+                if(skip_out_of_bounds)
+                {
+                    if(not output_shape.check_within_bounds(out_idx))
+                    {
+                        return;
+                    }
+                }
+                else
+                {
+                    assert(output_shape.check_within_bounds(out_idx));
```

Comment:
Put this assert before the if(skip_out_of_bounds) for release it shouldn't do anything, wheras will trigger on debug, and allow you to not need the else case which would be empty

Reply:
Similar to above, use the skip_out_of_bounds_blag

`assert(skip_out_of_bounds or output_shape.check_within_bounds(out_idx));` 

That eliminates the else branch